### PR TITLE
fix: [CDS-81646]: Log HARNESS_RESOURCE_CLASSES when register resources throws an exception

### PR DESCRIPTION
**Pull request by yogesh-chauhan**
- fix: [CDS-81646]: Log HARNESS_RESOURCE_CLASSES when register resources throws an exception
- fix: [CDS-81646]: Fix some code smells

## Describe your changes
Log HARNESS_RESOURCE_CLASSES when register resources throws an exception
This will help us to get the exact order of resource classes which leads to Guice exception during injection
image

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

[CDS-81646]: https://harness.atlassian.net/browse/CDS-81646?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CDS-81646]: https://harness.atlassian.net/browse/CDS-81646?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54091)

### CODE REVIEW

File: 120-ng-manager/src/main/java/io/harness/ng/NextGenApplication.java [hunk 1]
```diff
@@ -978,8 +984,7 @@ private void registerEtagFilter(Environment environment, Injector injector) {
   }
 
   private void registerWaitEnginePublishers(Injector injector) {
-    final QueuePublisher<NotifyEvent> publisher =
-        injector.getInstance(Key.get(new TypeLiteral<QueuePublisher<NotifyEvent>>() {}));
+    final QueuePublisher<NotifyEvent> publisher = injector.getInstance(Key.get(new TypeLiteral<>() {}));
```
Comment (yogesh-chauhan) [inline 120-ng-manager/src/main/java/io/harness/ng/NextGenApplication.java:987]: This is a code smell, explicit type arguments are not needed here

File: 120-ng-manager/src/main/java/io/harness/ng/NextGenApplication.java [hunk 2]
```diff
@@ -573,7 +574,7 @@ private void registerMigrations(Injector injector) {
   private NGMigrationConfiguration getMigrationSdkConfiguration() {
     return NGMigrationConfiguration.builder()
         .microservice(Microservice.CORE)
-        .migrationProviderList(new ArrayList<Class<? extends MigrationProvider>>() {
+        .migrationProviderList(new ArrayList<>() {
```
Comment (yogesh-chauhan) [inline 120-ng-manager/src/main/java/io/harness/ng/NextGenApplication.java:577]: This is a code smell, explicit type arguments are not needed here

File: 120-ng-manager/src/main/java/io/harness/ng/NextGenApplication.java [hunk 3]
```diff
@@ -508,7 +509,7 @@ public Class<? extends RemoteObserverInformer> getRemoteObserverImpl() {
 
     log.info("NextGenApplication DEPLOY_VERSION = " + System.getenv().get(DEPLOY_VERSION));
     if (DeployVariant.isCommunity(System.getenv().get(DEPLOY_VERSION))) {
-      initializeNGMonitoring(appConfig, injector);
+      initializeNGMonitoring(injector);
```
Comment (yogesh-chauhan) [inline 120-ng-manager/src/main/java/io/harness/ng/NextGenApplication.java:512]: This is a code smell, appConfig is not needed here

**Review by kapilgarg1996 — DISMISSED** [review]

**Review by PragyeshMishra01 — APPROVED** [review]

**Review by abhinavmittal-harness — DISMISSED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/54091)* status: 1 file(s) to review.

**Comment by yogesh-chauhan** [comment]
trigger uts

**Comment by yogesh-chauhan** [comment]
trigger ut0, ut3, ut4, ut8, ut9, review

**Comment by yogesh-chauhan** [comment]
trigger uts

**Review by abhinavsingh1196 — APPROVED** [review]

**Review by sarthakkasat — APPROVED** [review]

**Pull request merged**